### PR TITLE
Fix sheet name escaping (fixes #576)

### LINKED ISSUE #576: Incorrect escaping used for Excel and ODS spreadsheet names
In Excel, "test" is a valid spreadsheet name, and this needs to be saved in workbook.xml as

    <sheets><sheet name="&quot;test&quot;" sheetId="1" r:id="rId1"/></sheets>

However, this library gets that wrong. That is, this comment: https://github.com/box/spout/blob/master/src/Spout/Common/Escaper/XLSX.php#L45 and the following code is incorrect.

Ditto for ODS.

I am about to make a pull request for this.

### PR DESCRIPTION
**Pull request by timhunt**
I hope this is OK. This is the first time I have contributed to this project. Please let me know if I need to change anything. Thanks in advance for considering this.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @timhunt has signed the CLA. Thanks for the pull request!

File: src/Spout/Common/Escaper/XLSX.php [hunk 1]
```diff
@@ -42,9 +42,9 @@ protected function init()
     public function escape($string)
     {
         $escapedString = $this->escapeControlCharacters($string);
-        // @NOTE: Using ENT_NOQUOTES as only XML entities ('<', '>', '&') need to be encoded.
-        //        Single and double quotes can be left as is.
-        $escapedString = htmlspecialchars($escapedString, ENT_NOQUOTES, 'UTF-8');
+        // @NOTE: We must use ENT_QUOTES because single and double quotes may be used in spreadsheet names.
```
Comment (adrilo) [inline src/Spout/Common/Escaper/XLSX.php]: Can you please change that note to something more global like this: "Using ENT_QUOTES as XML entities ('<', '>', '&') as well as double quotes (for values of XML attributes) need to be encoded." ?

File: src/Spout/Common/Escaper/ODS.php [hunk 2]
```diff
@@ -26,13 +26,13 @@ public function escape($string)
...
             // @link https://github.com/box/spout/issues/329
-            $replacedString = htmlspecialchars($string, ENT_NOQUOTES | ENT_DISALLOWED, 'UTF-8');
+            $replacedString = htmlspecialchars($string, ENT_QUOTES | ENT_DISALLOWED, 'UTF-8');
         } else {
             // We are on hhvm or any other engine that does not support ENT_DISALLOWED.
             //
-            // @NOTE: Using ENT_NOQUOTES as only XML entities ('<', '>', '&') need to be encoded.
-            //        Single and double quotes can be left as is.
-            $escapedString =  htmlspecialchars($string, ENT_NOQUOTES, 'UTF-8');
+            // @NOTE: We must use ENT_QUOTES because single and double quotes may be used in spreadsheet names.
```
Comment (adrilo) [inline src/Spout/Common/Escaper/ODS.php]: Can you please change that note to something more global like this: "Using ENT_QUOTES as XML entities ('<', '>', '&') as well as double quotes (for values of XML attributes) need to be encoded." ?

**Comment by JasonEverling** [comment]
So can this bug get merged? It is affecting other software that use this library

**Comment by adrilo** [comment]
Actually, looking at the test changes, this change is not ok: `'adam\'s "car"'` being transformed into `'adam&#039;s &quot;car&quot;'` is not something we want. We need to find another way to fix this issue. Maybe have different escape methods?

**Comment by adrilo** [comment]
Closing this PR as I pushed another one here: #620 (was easier than resolving conflicts on your branch).

**Pull request closed without merging**